syntax = "proto3";
package VRDAVis;

import "defs.proto";

// FILE_INFO_REQUEST:
// Requests the file info for a specific file.
// Backend responds with FILE_INFO_RESPONSE
message FileInfoRequest {
    // Required directory name
    string directory = 1;
    // Required file name
    string file = 2;
}

// FILE_INFO_RESPONSE
// Response for FILE_INFO_REQUEST.
// Gives information on the requested file
message FileInfoResponse {
    // Defines whether the FILE_INFO_REQUEST was successful
    bool success = 1;
    // Error message (if applicable)
    string message = 2;
    // Basic file info (type, size)
    FileInfo file_info = 3;
}